src/coreclr/src/gc/gc.cpp

```diff
@@ -33940,7 +33837,7 @@ void gc_heap::process_n_background_segments (heap_segment* seg,
                                              heap_segment* prev_seg,
                                              generation* gen)
 {
-    assert (gen != large_object_generation);
+    assert (gen->gen_num <= max_generation);
```

Comment:
I'm ok with just deleting the `process_n_background_segments` method as it's unused